### DISCUSSION TITLE
Why isn't ikvm an iron language?

### DISCUSSION THREAD

**Discussion by kevincrans** [ikvmnet/ikvm#250] [Ideas]
This would be using the dlr and ngen to jit translate java bytecode to ngen binary(skipping clr for speed).

The first problem would be bytecode not being a language.
Would this work though?

I'm just new here, but I'm also curious.

**Comment by wasabii** [comment]
The DLR is a wholly inappropriate tool for designing a JVM. I'm not sure what not more to say. The DLR is for dynamic languages. A JVM isn't that.

**Reply by kevincrans** [reply]
Thank you for clarifying!
I'd argue that Ruby also isn't, but that case is very different.
Does that wipe ngen off the chart?

**Comment by wasabii** [comment]
I must be really confused. What does the DLR have to do with ngen anyways?

And ngen worked fine on IKVM assemblies last I checked.

**Reply by kevincrans** [reply]
True, I assume that cil, especially with the dlr would make it extra slow, that's where ngen jumps in. So I guess ngen seperately isn't related to this discussion indeed.
Happy new year anyway!

**Reply by wasabii** [reply]
IKVM produces IL just like any .NET code, and runs just as fast as it would doing the same task.

**Reply by kevincrans** [reply]
Ah, sometimes hard to find articles about this, thx for the better observation.
